generator/tree.go

```diff
@@ -205,6 +205,34 @@ func generateConfigModule(cfg *ModuleConfig, node *Node, nameToNode map[string]*
 	needToWalk := map[string]struct{}{}
 	tableInstances := map[string][]string{}
 
+	// Apply type overrides for the current module.
+	newTree := false
+	for name, params := range cfg.Overrides {
+		if params.Type == "" {
+			continue
+		}
+		// Make a copy of the tree if not done yet.
```

Comment:
Hmm, I think this would be simplest to always be done in the function that calls this one. Performance isn't the primary goal of the generator, it's a rarely run batch process.